cli/cmd/install.go

```diff
@@ -1108,3 +1168,25 @@ func toIdentityContext(idvals *identityWithAnchorsAndTrustDomain) *pb.IdentityCo
 		Scheme:             idvals.Identity.Issuer.Scheme,
 	}
 }
+func checkAddons(values *l5dcharts.Values) []l5dcharts.AddOn {
+	var addons []l5dcharts.AddOn
+
+	if values.Tracing != nil {
+		if values.Tracing.Enabled {
+			addons = append(addons, values.Tracing)
+		}
+	}
+
+	return addons
+}
+
+func mergeAddonValues(values, addonValues *l5dcharts.Values) error {
+
+	if addonValues.Tracing != nil {
+		if err := mergo.Merge(addonValues.Tracing, values.Tracing); err != nil {
+			return err
+		}
+		values.Tracing = addonValues.Tracing
+	}
+	return nil
+}
```

Comment:
Why not having the new `--config` flag serve as overrides for all the values, not just for add-ons? And if so, it might be clearer to implement that as a preliminary PR...

Reply:
The main reason is that, some fields in `values` like `config` etc are generated way before in the install/upgrade path but as we override them with config last in the pipeline, this would require us to again redo those values. 

Sure, we can keep this overwrite logic in the start but that would require a lot of code-changes. So, the plan is to do that next but keep it out of this PR. WDYT?

Reply:
I'm not sure I follow. Doesn't this overriding logic imply that only the values present in the file provided through `--config` will get overridden, leaving all the others untouched? Why would it need to regenerate those other values? Maybe an example might help me understand :slightly_smiling_face:

Reply:
Ah, Sorry for not being clear.
My main concern is with this, as it is built by combining various values and is added in the `linkerd-config` cm, we will have to regenerate that agian after the override this. We could regenrate that again if it was based on `Values` struct but currently it is based on `identityInstallOptions`.
 https://github.com/linkerd/linkerd2/blob/5fc83bc1c1b81487172364548a46be80618aa4ad/pkg/charts/linkerd2/values.go#L41

Reply:
Gotcha. Another reason to refactor this at some point in the future  so we have to rely less on the structs in install.go and leverage those in values.go more directly :wink:

Reply:
I recommend excluding this change from this PR for now. It feels like a more complicated problem than we first thought. The addon will still work without the `--config` flag, right? It will be restricted to just defaults in the `values.yaml`.